onmetal-controller-manager/main.go

```diff
@@ -364,6 +365,22 @@ func main() {
 		}
 	}
 
+	if controllers.Enabled(natGatewayController) {
+		if err = onmetalapiclient.SetupNATGatewayNetworkNameFieldIndexer(context.TODO(), mgr.GetFieldIndexer()); err != nil {
+			setupLog.Error(err, "unable to setup field indexer", "field", onmetalapiclient.NATGatewayNetworkNameField)
+			os.Exit(1)
+		}
+
+		if err = (&networkingcontrollers.NatGatewayReconciler{
+			Client:        mgr.GetClient(),
+			Scheme:        mgr.GetScheme(),
+			EventRecorder: mgr.GetEventRecorderFor("nat-gateway-scheduler"),
```

Comment:
```suggestion
			EventRecorder: mgr.GetEventRecorderFor("natgateways"),
```
The name `scheduler` might be misleading since we don't schedule `NATGateway`s anywhere.
Also, that naming would be more in-line with the naming of our existing `EventRecorder`s.